### PR TITLE
xrCore: application title member added

### PR DESCRIPTION
**Pull request by vTurbine**
The only leftover to completely fill `VkApplicationInfo`

### CODE REVIEW

File: src/xrCore/xrCore.h [hunk 1]
```diff
@@ -142,6 +142,7 @@ class XRCORE_API xrCore
 public:
     xrCore();
 
+    string64 ApplicatiionTitle;
```
Comment (Xottab-DUTY) [inline src/xrCore/xrCore.h]: Typo
```suggestion
    string64 ApplicationTitle;
```

File: src/xrEngine/Device_Initialize.cpp [hunk 2]
```diff
@@ -66,7 +66,9 @@ void CRenderDevice::Initialize()
 
         pcstr title = READ_IF_EXISTS(pSettingsOpenXRay, r_string, "window", 
                                     "title", "S.T.A.L.K.E.R.: Call of Pripyat");
-        
+
+        VERIFY(xr_strlen(title) < sizeof(Core.ApplicatiionTitle));
+        xr_strcpy(Core.ApplicatiionTitle, title);
```
Comment (Xottab-DUTY) [inline src/xrEngine/Device_Initialize.cpp]: Typo
```suggestion
        xr_strcpy(Core.ApplicationTitle, title);
```

File: src/xrEngine/Device_Initialize.cpp [hunk 3]
```diff
@@ -66,7 +66,9 @@ void CRenderDevice::Initialize()
 
         pcstr title = READ_IF_EXISTS(pSettingsOpenXRay, r_string, "window", 
                                     "title", "S.T.A.L.K.E.R.: Call of Pripyat");
-        
+
+        VERIFY(xr_strlen(title) < sizeof(Core.ApplicatiionTitle));
```
Comment (Xottab-DUTY) [inline src/xrEngine/Device_Initialize.cpp]: It's better to say:
```suggestion
        if (xr_strlen(title) < std::size(Core.ApplicationTitle))
            Log("~ Your application title is too big to fit in Core.ApplicationTitle. Shrink it to 64 characters.");
```

**Review by Xottab-DUTY — CHANGES_REQUESTED** [review]

File: src/xrEngine/Device_Initialize.cpp [hunk 4]
```diff
@@ -66,7 +66,8 @@ void CRenderDevice::Initialize()
 
         pcstr title = READ_IF_EXISTS(pSettingsOpenXRay, r_string, "window", 
                                     "title", "S.T.A.L.K.E.R.: Call of Pripyat");
-        
+
+        strncpy_s(Core.ApplicationTitle, title, sizeof(Core.ApplicationTitle) - 1);
```
Comment (Xottab-DUTY) [inline src/xrEngine/Device_Initialize.cpp]: `xr_strcpy` is save and it's enough, hence there's no need to use `strncpy_s`.
```suggestion
        xr_strcpy(Core.ApplicationTitle, title);
```

**Review by Xottab-DUTY — CHANGES_REQUESTED** [review]

File: src/xrEngine/Device_Initialize.cpp (hunk 4 above)
Comment (vTurbine) [reply src/xrEngine/Device_Initialize.cpp]: Indeed. Thank you

**Comment by vTurbine** [comment]
Decided do not add any log messages related to size mismatch since this title is for internal use only

File: src/xrCore/xrCore.h [hunk 5]
```diff
@@ -142,6 +142,7 @@ class XRCORE_API xrCore
 public:
     xrCore();
 
+    string64 ApplicationTitle;
```
Comment (Xottab-DUTY) [inline src/xrCore/xrCore.h]: Uhm, sorry that I didn't noticed it earlier:
Logically, `ApplicationTitle` should be lower than `ApplicationName`.

**Comment by Xottab-DUTY** [comment]
> Decided do not add any log messages related to size mismatch since this title is for internal use only

Yes, that's ok.

File: src/xrCore/xrCore.h (hunk 5 above)
Comment (vTurbine) [reply src/xrCore/xrCore.h]:
```
string64 ApplicationName;
string64 ApplicationTitle;
```
?

Comment (Xottab-DUTY) [reply src/xrCore/xrCore.h]: Yes

**Review by Xottab-DUTY — APPROVED** [review]

**Comment by Xottab-DUTY** [comment]
Thank you!

**Pull request merged**